### DISCUSSION TITLE
Problem with video-js and hls on Linux

### DISCUSSION THREAD

**Discussion by HennHoff** [tauri-apps/tauri#6766] [Q&A]
I'm trying to play a hls stream with video-js.
Running my code with Tauri under Windows works just fine. Running it from a webserver in Chromium or Firefox under Linux also works just fine.
Running it with Tauri under Linux the video is stuck on loading. Comparing the debug messages from video-js it seems there is an issue with loading/appending the audio stream.
I don't get any errors or warnings and have run out of ideas how to fix this. I'm hoping someone can help me or point me in a direction to resolve this.

[Here is my test code](https://github.com/HennHoff/test-tauri-videojs), it should play the most popular video in the US region on YouTube through Piped.

**Comment by FabianLars** [comment]
Just to check if it's Tauri specific or webkitgtk in general could you try the same thing in epiphany (also known as Gnome web)? It's a browser using webkitgtk so it will more closely resemble tauri than firefox or chrome.

**Comment by HennHoff** [comment]
Thanks for the fast answer.
With Epiphany it doesn't work either. So it seems it's a problem with webkitgtk in general.
I will ask around there :)

**Reply by angelinimattia** [reply]
Hi I'm trying to solve the same issue, I tried using wasp ( an hls integration as web assembly ) but no luck in that since tauri do not support some features that wasp is using on web assembly. Tried other adaptive protocols as dash but still no luck. Did you mange to make it work.

**Comment by angelinimattia** [comment]
I manage to make it work by removing video-js and using the video element:
```html
 <video controls >
        <source
                src="https://demo.unified-streaming.com/k8s/features/stable/video/tears-of-steel/tears-of-steel.ism/.m3u8"
                type="application/x-mpegURL"
        />
</video> 
```
